.husky/commit-msg

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx --no-install commitlint --edit "$1"
```

Comment:
Given that you're already installing commitlint, I'd recommend adding an npm script for commitlint, and then calling it from the locally installed bin.

In **package.json**:
```json
//...
  "scripts": {
    "commitlint": "commitlint"
  }
//...
```

In `.husky/commit-msg`:
```bash
npm run commitlint --edit "$1"
```

Alternatively, you can make husky-specific scripts and manage them with the rest of your scripts like so:
 - husky config: https://github.com/emulsify-ds/emulsify-cli/blob/develop/.husky/commit-msg
 - npm script: https://github.com/emulsify-ds/emulsify-cli/blob/develop/package.json#L19

Reply:
These `.husky/*` files are created by husky following their documentation: https://typicode.github.io/husky/#/?id=create-a-hook
And Commitlint has this in their docs: https://github.com/conventional-changelog/commitlint#getting-started

I'm can't answer why they do it this way vs an npm script like you recommend, but it's their "official" way...

Reply:
@ModulesUnraveled I recommend using the locally-installed version of commitlint, because it'll lock you into a specific version, and allow you to make sure upgrades are coupled with any needed config changes, etc. It also ensures that all developers are using the same version of commitlint.

Reply:
That makes sense to me! I'll update it.

Any idea why they don't have that in the docs?

Reply:
Oh, and if we're using yarn for everything should I use that in the husky commands as well? so

```
yarn commitlint --edit "$1"
```

instead of `npm run`?

Reply:
> Any idea why they don't have that in the docs?

I'm not sure. It works, and perhaps is the path of least resistance, especially since yarn/npm are slightly different?

> Oh, and if we're using yarn for everything should I use that in the husky commands as well?

Yep! I don't personally have a preference between the two package managers, but if you pick one, you wanna buy into it across the board and avoid using both npm and yarn in the same project.

Reply:
👍  Thanks!